src/main/java/org/sagebionetworks/web/server/servlet/filter/HtmlInjectionFilter.java

```diff
@@ -149,6 +177,42 @@ public static boolean isGWTPlace(String targetPath) {
     //        return false;
   }
 
+  private void addCdnEndpoint(
+    Map<String, String> dataModel,
+    HttpServletRequest request
+  ) {
+    String origin = request.getHeader("origin");
+    if (origin != null && CDN_ORIGINS_REGEX.matcher(origin).matches()) {
+      dataModel.put(CDN_ENDPOINT_KEY, "//cdn-" + request.getHeader("origin"));
+    } else {
+      dataModel.put(CDN_ENDPOINT_KEY, "");
+    }
+  }
```

Comment:
Moved logic to determine the CDN endpoint to this filter, instead of JavaScript in the HTML file itself.  Logic should be the same.